rosidl_parser/rosidl_parser/definition.py

```diff
@@ -107,170 +167,231 @@ def __eq__(self, other):
         return super().__eq__(other) and self.name == other.name
 
 
-class NamespacedType(BaseType):
+class NamespacedType(AbstractNestableType):
     """A type identified by a name in a namespaced scope."""
 
     __slots__ = ('namespaces', 'name')
 
-    def __init__(self, namespaces, name):
+    def __init__(self, namespaces: Iterable[str], name: str):
         """
         Constructor.
 
-        :param list[str] namespaces: the names of nested namespaces identifying
-          a specific scope
-        :param str name: the name of the type within that scope
+        :param namespaces: the names of nested namespaces identifying a
+          specific scope
+        :param name: the name of the type within that scope
         """
         super().__init__()
         self.namespaces = namespaces
         self.name = name
 
+    def namespaced_name(self) -> Tuple[str, ...]:
+        return (*self.namespaces, self.name)
```

Comment:
Since this is used together in a lot of places providing the combination in a method makes sense.

Reply:
Depending on how this is used, consider making this a property, adding a little sugar.

Reply:
I would lean towards keeping it a method so the caller sees what is actually an attribute and what is derived data. If others too think it should be a property I am happy to change it though.